services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyNoVncClient.java

```diff
@@ -284,6 +285,7 @@ private void connectClientToVNCServer(String tunnelUrl, String tunnelSession, St
         try {
             if (StringUtils.isNotBlank(websocketUrl)) {
                 s_logger.info(String.format("Connect to VNC over websocket URL: %s", websocketUrl));
+                ConsoleProxy.ensureRoute(NetUtils.extractHost(websocketUrl));
```

Comment:
@DaanHoogland 
`UriUtils` seems more suitable than `NetUtils`

Reply:
I checked but UriUtils only deals with the class URI. I can move it though, I don't think it matters much.